### DISCUSSION TITLE
[archived] "Workflow Controller ConfigMap" docs discussion

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#9635] [General]
# argo-workflows/workflow-controller-configmap/

https://argoproj.github.io/argo-workflows/workflow-controller-configmap/

**Comment by ghost** [comment]
Is there a way to send the logs to a particular folder/path within the S3 bucket?

**Reply by ghost** [reply]
I see it is possible by passing `keyFormat`

**Reply by nicolas-vivot** [reply]
it is, but know that there is a weird behavior with that:
- if you do not override path for output artifacts, by default it would take the name of the artifacts and store it under the default path specified under keyFormat.
- if you specify a path in the output artifacts, it will actually override the default value of keyFormat, which means you need to include what you have in keyFormat every where you override path for output artifacts.

So it sounds more like a default value than a forced prefix (as documented)
Not sure if this is intended but incorrectly documented, or a bug.
Definitely for me it's a bug and i would like to have the keyFormat be a forced prefix when using a specific repository rather than having to include that prefix everytime we redefine artifacts path.

**Comment by dvaldivia** [comment]
How can we add volumes to the workflow? for example, if we need the workflow to trust additional certificates by default for the s3 endpoint

**Reply by agilgur5** [reply]
I'm not sure if this is related to this page per se (this page is for global configuration of Argo for cluster operators), but there are a couple of ways to use volumes in workflows.

See the [Volumes doc](https://argoproj.github.io/argo-workflows/walk-through/volumes/) for a few examples.
You could also use that together with [`workflowDefaults`](https://argoproj.github.io/argo-workflows/default-workflow-specs/#setting-default-workflow-values).

**Comment by Nicholas-Laughlin** [comment]
Hello! If i wanted to create a workflow template that allows me connect to a repo in github, then run other workflows on that repo. are there any examples of this or could you provide insight?

**Reply by agilgur5** [reply]
This doesn't sound related to this page (this page is for global configuration of Argo for cluster operators), but yes, Workflows can create other Workflows.

See the [Workflow of Workflows pattern](https://argoproj.github.io/argo-workflows/workflow-of-workflows/) for an example.
There is also built-in support for [using `git` as an artifact](https://argoproj.github.io/argo-workflows/walk-through/hardwired-artifacts/).

**Comment by pedrovgp** [comment]
I want to use Katib to optimize argo workflows. That requires to set the [runtimeContainerExecutor to emissary](https://github.com/kubeflow/katib/tree/master/examples/v1beta1/argo#argo-workflow).

But I have other workflows that require it not to be set. Is there a way to set the containerRuntimeExecutor only for Katib Experiments, maybe in the API yaml file?

**Reply by agilgur5** [reply]
Emissary is the only supported runtime executor in Argo 3.4+. But in Argo 3.3 and below, you can use [`containerRuntimeExecutors`](https://github.com/argoproj/argo-workflows/blob/30a916880ff21027ad1004c4ae176037f1a956da/docs/workflow-controller-configmap.yaml#L167) (note the plural `s`) which allows you to specify a `selector` for each runtime executor.

**Comment by DaraDadachanji** [comment]
If I'm doing a managed namespace install, do I need to add a workflow-controller-configmap to each namespace where I am running workflows? Or is there a way to have it use the one in the namespace where argo is installed?

**Reply by agilgur5** [reply]
In the same namespace that Argo is installed. The ConfigMap affects the Controller globally.
